src/main/java/seedu/toluist/controller/FindController.java

```diff
@@ -0,0 +1,163 @@
+    private static final String SPLITTER_COMMAND = " ";
+      
+    private final Logger logger = LogsCenter.getLogger(getClass());
+
+    public FindController(Ui renderer) {
+        super(renderer);
+    }
+
+    public CommandResult execute(String command) {
+        logger.info(getClass() + "will handle command");
+        
+        //initialize variables for searching
+        ArrayList<Task> foundTasks = new ArrayList<Task>();
+        HashMap<String, String> tokens = tokenize(command);
+        boolean isSearchByTag = tokens.get(SEARCH_BY_TAG).equals(TRUE_COMMAND);
+        boolean isSearchByName = tokens.get(SEARCH_BY_NAME).equals(TRUE_COMMAND);
+        String keywords = tokens.get(KEYWORDS);
+        String[] keywordList = keywords.split(" ");
+        keywordList = removeEmptySpaces(keywordList);
+        boolean isFound = false;
+        TodoList todoList = TodoList.load();
+        ArrayList<Task> taskList = todoList.getTasks();
+        Task currentTask;
+        int foundValue = 0;
+
+        for (int i = 0; i < taskList.size(); i++) {
+            currentTask = taskList.get(i);
+            for (int j = 0; j < keywordList.length; j++) {
+                if (isSearchByTag && !isFound) {
+                    for (int k = 0; k < currentTask.allTags.size(); k++) {
+                        if (currentTask.allTags.get(k).tagName.toLowerCase()
+                                .contains(keywordList[j].toLowerCase())) {
+                            foundTasks.add(currentTask);
+                            isFound = true;
+                            foundValue++;
+                        }   
+                    }                    
+                }
+                if (isSearchByName && !isFound) {
+                        if (currentTask.description.toLowerCase()
+                            .contains(keywordList[j].toLowerCase())) {
+                        foundTasks.add(currentTask);
+                        isFound = true;
+                        foundValue++;
+                    }                
+                }
+            }
+            isFound = false;
+        }
+        
+        uiStore.setTask(foundTasks);
+        renderer.render();
+
+        //display formatting
+        return formatDisplay(isSearchByTag, isSearchByName, keywordList, foundValue);
+    }
+
+    private String[] removeEmptySpaces(String[] keywordList) {
+        ArrayList<String> replacementList = new ArrayList<String>();
+        for (int i = 0; i < keywordList.length; i++) {
+            if (!keywordList[i].equals("")) {
+                replacementList.add(keywordList[i]);
+            }
+        }
+        return replacementList.toArray(new String[0]);
+    }
+
+    private CommandResult formatDisplay(boolean isSearchByTag, boolean isSearchByName,
+                                        String[] keywordList, int foundValue) {
+        String searchParameters;
+        if (isSearchByName) {
+            if (isSearchByTag) {
```

Comment:
Make user of terniary operator to make your code more concise. eg. searchParameter = isSearchByTag ? ... : ...